pkg/apply/taskrunner/context.go

```diff
@@ -53,6 +53,12 @@ func (tc *TaskContext) ResourceCache() cache.ResourceCache {
 	return tc.resourceCache
 }
 
+// SendEvent sends an event on the event channel
+func (tc *TaskContext) SendEvent(e event.Event) {
```

Comment:
I like this new method, it's a good idea. I wonder if we can delete the `EventChannel()` method to encapsulate the channel even more? No need to do it in this PR, of course.

Reply:
The EventChannel method is used in a few places to read from the event channel, like in the ApplyTask which sends it to `applyOptionsFactoryFunc` (aka `newApplyOptions`) which passes it to `KubectlPrinterAdapter`. I don't really understand why it's building a printer, but if you wanna clean it up, be my guest.

I mostly just added the SendEvent convenience method so there was a single place to log from.